packages/chat_repository_interface/lib/src/interfaces/pending_message_repository_interface.dart

```diff
@@ -27,6 +29,7 @@ abstract class PendingMessageRepositoryInterface {
     String? imageUrl,
     String? messageType,
     DateTime? timestamp,
+    Uint8List? imageData,
```

Comment:
Let's not introduce a different datatype to represent images, but rather use the imageUrl with an additional type so we can support multiple types. If we base64encode the image, we can both store the image as url or as data in the same type.

Reply:
Because both imageUrl and type are nullable, I would introduce a new internal messageImage class, that would include both parameters and make only the messageImage nullable on the class.